tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckDates.java

```diff
@@ -821,18 +819,28 @@ public static String checkIso8601(String path, String value) {
             VariableField field = (VariableField) p;
             int type = field.getType();
             if (!expectedField.contains(type)) {
-                return "Disallowed field: " + field;
+                return "Field " + field + " is not allowed in a YMD calendar.";
             }
             // The two parts of an interval are identified by when you hit the same type of field
             // twice
             // like y - y, or M d - M
             if (fieldTypesSoFar.contains(type)) {
-                if (isInterval) { // so one freebe for intervals
-                    isInterval = false;
+                if (isInterval && intervalPosition.equals("1st")) { // so one freebe for intervals
+                    intervalPosition = "2nd";
                     fieldTypesSoFar.clear(); // entering second part of interval
                     lastField = null;
                 } else {
-                    return "Duplicate field: " + field;
+                    return "Field " + field + " is the same type as a previous field.";
+                }
+            }
+
+            // No year truncation
+
+            if (type == DateTimePatternGenerator.YEAR) {
+                if (field.toString().length() == 2) {
+                    return "Field "
+                            + field
+                            + " is incorrect. For a YMD calendar, the year field cannot be truncated to 2 digits.";
```

Comment:
```suggestion
                    return "Field "
                            + field
                            + " is incorrect. For a YMD calendar, the year field must be a single 'y'.";
```

Shouldn't it be always 'y'?

Reply:
No, someone could have yyyy (minimum 4 digits), which is actually better for gregorian (but only slightly; not that many people's dates are before 1000 AD)